wahoomc/main.py

```diff
@@ -63,6 +64,9 @@ def run(run_level):
         elif o_input_data.xy_coordinates:
             o_osm_data = XYOsmData(o_input_data)
 
+        log = logging.getLogger('main-logger')
```

Comment:
I tried to not to log in the main.py file.

Thinking of the idea with the class, it might be thinkable to let this class also do logging.
At least the unique logging you do in this file, line 110. You could write a general method where you give the text, i.e. "total time" and always add the deltas of walltime and CPU at the end.